File: Selenium/Project/Activity7.py
# Import webdriver from selenium
import pytest
import time

from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())
# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Open the browser to the URL
    driver.get("https://alchemy.hguy.co/jobs")
    print("Page Title is ", driver.title)
    driver.implicitly_wait(10)
    driver.find_element(By.XPATH,"//a[contains(text(),'Post a Job')]").click()
    driver.find_element(By.XPATH,"//a[contains(text(),'Sign in')]").click()
    driver.find_element(By.ID,"user_login").clear()
    driver.find_element(By.ID,"user_login").send_keys("root")
    driver.find_element(By.ID,"user_pass").clear()
    driver.find_element(By.ID,"user_pass").send_keys("pa$$w0rd")
    driver.find_element(By.XPATH,"//input[@id='wp-submit']").click()
    driver.find_element(By.XPATH,"//input[@id='job_title']").send_keys("Banking")
    driver.find_element(By.XPATH,"//input[@id='job_location']").send_keys("India")
    driver.find_element(By.XPATH,"//iframe[@id='job_description_ifr']").send_keys("Description")
    driver.find_element(By.XPATH,"//input[@id='company_name']").clear()
    driver.find_element(By.XPATH,"//input[@id='company_name']").send_keys("ABC")
    driver.find_element(By.XPATH,"//body/div[@id='page']/div[@id='content']/div[1]/div[1]/main[1]/article[1]/div[1]/form[1]/p[1]/input[4]").click()
    driver.find_element(By.XPATH,"//input[@id='job_preview_submit_button']").click()
    print("Job Submitted Successfully")

    driver.close()

File: Selenium/Project/Activity7.java
package JOBBOARD;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

public class Activity7 {
    WebDriver driver;

    @BeforeClass
    public void Beforeclass() {
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriverManager.firefoxdriver().setup();
        driver = new FirefoxDriver();
        driver.get("https://alchemy.hguy.co/jobs");
    }

    @Test
    public void CreateJob() throws InterruptedException {
        driver.findElement(By.xpath("//a[contains(text(),'Post a Job')]")).click();
        Thread.sleep(1000);
        driver.findElement(By.xpath("//a[contains(text(),'Sign in')]")).click();
        driver.findElement(By.id("user_login")).sendKeys("root");
        driver.findElement(By.id("user_pass")).sendKeys("pa$$w0rd");
        driver.findElement(By.xpath("//input[@id='wp-submit']")).click();
        driver.findElement(By.xpath("//input[@id='job_title']")).sendKeys("Banking");
        driver.findElement(By.xpath("//input[@id='job_location']")).sendKeys("India");
        driver.findElement(By.xpath("//iframe[@id='job_description_ifr']")).sendKeys("Description");
        driver.findElement(By.xpath("//input[@id='company_name']")).clear();
        driver.findElement(By.xpath("//input[@id='company_name']")).sendKeys("ABC");
        driver.findElement(By.xpath("//body/div[@id='page']/div[@id='content']/div[1]/div[1]/main[1]/article[1]/div[1]/form[1]/p[1]/input[4]")).click();
        Thread.sleep(1500);
        driver.findElement(By.xpath("//input[@id='job_preview_submit_button']")).click();
        //Verify that the job listing was posted by visiting the jobs pag
        driver.close();



    }
}
